galaxyui/src/app/authors/detail/author-detail.component.ts

```diff
@@ -356,7 +369,7 @@ export class AuthorDetailComponent implements OnInit {
     }
 
     private prepareRepositories() {
-        this.items.forEach((item: Repository) => {
+        this.repositories.forEach((item: Repository) => {
```

Comment:
Is there no similar need for a `prepareCollections()` helper?

Reply:
I really hate this pattern because the `prepareRepositories()` function adds new attribute to the `Repository` type. The result is that you end up with a list of repositories that don't actually match the correct type definition and that makes it a lot hard to debug and extend the UI. If I had more time I'd go back and fix the legacy code on this page so that it doesn't mess with the `Repository` type.

We also don't really need a `prepareCollections()` function because the backend already gives us all the data we need for displaying collections.